### DISCUSSION TITLE
How to manipulate input value

### DISCUSSION THREAD

**Discussion by AmodeusR** [react-hook-form/react-hook-form#8803] [Q&A]
I want to manipulate an input value so I can limit the max length of the string as format it in a specific way. I tried getting the value with the _watch_ function, but trying to assign its value to the input value just broke the input. How can I properly get the value to manipulate it and reattribute to the input value without breaking it?

**Comment by Moshyfawn** [comment]
What you're talking about is a controlled input. You should look into using the `Controller` component or even create a reusable one with the `useController` hook. `register` won't suite inputs that need their values manipulated.

With `Controller`, you can define your custom value formatting logic before you send the value to the `field.onChange` handler

**Comment by AmodeusR** [comment]
This controller component is crazy complicated... And the tutorial on the React Hook Form website doesn't help with that. Are there any comprehensible tutorial somewhere in how to use it?
